src/spaced/spaced/game/player.cpp

```diff
@@ -15,30 +15,28 @@ using bave::PointerTap;
 using bave::Seconds;
 using bave::Shader;
 
-Player::Player(Services const& services) : m_services(&services), m_controller(services) {
+Player::Player(Services const& services, std::unique_ptr<IController> controller) : m_services(&services), m_controller(std::move(controller)) {
```

Comment:
why the switch to `unique_ptr`? also, what is the "I" in "Icontroller"?

Reply:
I means "interface", that it's an abstract class with no implementation, and that's also why a `unique_ptr` is needed, since `Player` only cares about the interface, the concrete implementation is passed to it by the owner. Polymorphic dependency injection, in short.